### PR TITLE
Fix issue with x-* fields Unmarshal

### PR DESCRIPTION
**Pull request by sayboras**
Addressed https://github.com/getkin/kin-openapi/issues/147

### CODE REVIEW

**Comment by fenollp** [comment]
Could you add a test for this? Could you also return on non-nil `err` before mutating `props`?

**Comment by sayboras** [comment]
> Could you add a test for this? Could you also return on non-nil `err` before mutating `props`?

Thanks for your feedback. Let me make an update :)

**Comment by sayboras** [comment]
@fenollp I have added unit test cases, which I think is required.

File: openapi3/extension_test.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+				"field2": "value2",
+			},
+		}
+
+		var value = &struct {
+			Field1 string `json:"field1"`
+			Field2 string `json:"field2"`
+		}{}
+
+		err = extensionProps.DecodeWith(decoder, value)
```
Comment (fenollp) [inline openapi3/extension_test.go]: Please verify `value` in this and the following tests

File: openapi3/extension.go [hunk 2]
```diff
@@ -25,13 +25,15 @@ func (props *ExtensionProps) EncodeWith(encoder *jsoninfo.ObjectEncoder, value i
 
 // DecodeWith will be invoked by package "jsoninfo"
 func (props *ExtensionProps) DecodeWith(decoder *jsoninfo.ObjectDecoder, value interface{}) error {
+	err := decoder.DecodeStructFieldsAndExtensions(value)
+	if err != nil {
```
Comment (fenollp) [inline openapi3/extension.go]: You can write this in one line:
```golang
	if err := decoder.DecodeStructFieldsAndExtensions(value); err != nil {
```

File: go.mod [hunk 3]
```diff
@@ -8,3 +8,5 @@ require (
 	gopkg.in/check.v1 v1.0.0-20180628173108-788fd7840127 // indirect
 	gopkg.in/yaml.v2 v2.2.2 // indirect
 )
+
+go 1.13
```
Comment (fenollp) [inline go.mod]: This does not need to be in this PR. Please revert.

Comment (sayboras) [reply go.mod]: yeah, my IDE added this, and i committed the change by mistake :(

File: openapi3/extension_test.go (hunk 1 above)
Comment (sayboras) [reply openapi3/extension_test.go]: Thanks, I have added assert statement, another look if you don't mind 💯

File: openapi3/extension.go (hunk 2 above)
Comment (sayboras) [reply openapi3/extension.go]: Updated.
